src/details/ArborX_DetailsDistributor.hpp

```diff
@@ -168,24 +307,29 @@ class Distributor
         "");
 #endif
 
+    // If _permute is empty, we are assuming that we don't need to permute
+    // exports.
     Kokkos::View<ValueType *, typename View::traits::device_type> dest_buffer(
-        Kokkos::ViewAllocateWithoutInitializing("destination_buffer"),
-        exports.size());
-
-    // We need to create a local copy to avoid capturing a member variable
-    // (via the 'this' pointer) which we can't do using a KOKKOS_LAMBDA.
-    // Use KOKKOS_CLASS_LAMBDA when we require C++17.
-    auto const permute_copy = _permute;
-
-    Kokkos::parallel_for("copy_destinations_permuted",
-                         Kokkos::RangePolicy<ExecutionSpace>(
-                             0, _dest_offsets.back() * num_packets),
-                         KOKKOS_LAMBDA(int const k) {
-                           int const i = k / num_packets;
-                           int const j = k % num_packets;
-                           dest_buffer(num_packets * permute_copy[i] + j) =
-                               exports[num_packets * i + j];
-                         });
+        "destination_buffer", 0);
+    if (_permute.size() != 0)
```

Comment:
For clarity, I would probably introduce a bool variable with a clear name, something like `do_permute` to have `if (do_permute)`.

Reply:
Do you feel strongly about this?

Reply:
A bit, but we can look for a third opinion.